File: tenet-0/shared/bus-ts/src/status.ts
// Status strings returned by stored procedures. Centralized to prevent typos
// in switch statements from silently masking SP behavior.

export const spStatus = {
  ok: "ok",
  rejectedUnauthenticated: "rejected_unauthenticated",
  rejectedNamespace: "rejected_namespace",
  rejectedConstitution: "rejected_constitution",
  rejectedCausality: "rejected_causality",
  rejectedNoConstitution: "rejected_no_constitution",
} as const;

export const budgetStatus = {
  ok: "ok",
  warning: "warning",
  blocked: "blocked",
  unauthenticated: "unauthenticated",
} as const;

File: tenet-0/shared/bus-go/status.go
package bus

// Status strings returned by stored procedures. Keeping them centralized
// prevents typos in switch statements from silently masking SP behavior.
const (
	spStatusOK                      = "ok"
	spStatusRejectedUnauthenticated = "rejected_unauthenticated"
	spStatusRejectedNamespace       = "rejected_namespace"
	spStatusRejectedConstitution    = "rejected_constitution"
	spStatusRejectedCausality       = "rejected_causality"
	spStatusRejectedNoConstitution  = "rejected_no_constitution"

	budgetStatusOK              = "ok"
	budgetStatusWarning         = "warning"
	budgetStatusBlocked         = "blocked"
	budgetStatusUnauthenticated = "unauthenticated"
)
